File: user-service/src/main/java/rs/edu/raf/model/user/Role.java
package rs.edu.raf.model.user;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Getter
@NoArgsConstructor
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @ManyToMany(mappedBy = "roles")
    private List<User> users;

    public Role(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/model/user/Role.kt
package org.example.userservicekotlin.model.user

import jakarta.persistence.*

@Entity
class Role (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,
    val name: String,
    @ManyToMany(mappedBy = "roles")
    val users: List<User>
) {

    constructor(name: String): this(0, name, emptyList())

    override fun toString(): String {
        return name
    }
}
